include/nbl/builtin/hlsl/workgroup2/shared_scan.hlsl

```diff
@@ -233,37 +241,23 @@ struct reduce<Config, BinOp, 3, device_capabilities>
         using params_lv2_t = subgroup2::ArithmeticParams<config_t, BinOp, Config::ItemsPerInvocation_2, device_capabilities>;
         BinOp binop;
 
-        const uint32_t invocationIndex = workgroup::SubgroupContiguousIndex();
-        // level 0 scan
-        subgroup2::reduction<params_lv0_t> reduction0;
-        [unroll]
-        for (uint32_t idx = 0, virtualInvocationIndex = invocationIndex; idx < Config::VirtualWorkgroupSize / Config::WorkgroupSize; idx++)
-        {
-            vector_lv0_t scan_local;
-            dataAccessor.template get<vector_lv0_t>(idx * Config::WorkgroupSize + virtualInvocationIndex, scan_local);
-            scan_local = reduction0(scan_local);
-            if (Config::electLast())
-            {
-                const uint32_t virtualSubgroupID = Config::virtualSubgroupID(glsl::gl_SubgroupID(), idx);
-                const uint32_t bankedIndex = Config::sharedCoalescedIndexNextLevel(virtualSubgroupID, Config::ItemsPerInvocation_1);    // (virtualSubgroupID & (Config::ItemsPerInvocation_1-1)) * Config::SubgroupsPerVirtualWorkgroup + (virtualSubgroupID/Config::ItemsPerInvocation_1);
-                scratchAccessor.template set<scalar_t>(bankedIndex, scan_local[Config::ItemsPerInvocation_0-1]);   // set last element of subgroup scan (reduction) to level 1 scan
-            }
-        }
-        scratchAccessor.workgroupExecutionAndMemoryBarrier();
+        reduce<Config, BinOp, 2, device_capabilities>::template __doLevel0<DataAccessor, ScratchAccessor, params_lv0_t, vector_lv0_t>(dataAccessor, scratchAccessor);
 
+        const uint32_t invocationIndex = workgroup::SubgroupContiguousIndex();
         // level 1 scan
+        const uint32_t lv1_smem_size = Config::__ItemsPerVirtualWorkgroup;
```

Comment:
I'm pretty sure thats not correct, you write out `VirtualWorkgroupSize/SubgroupSize` scalars during the first pass

Reply:
ah the variable is named completely wrong.